src/lib.rs

```diff
@@ -417,40 +447,41 @@ impl HexView {
         for lines in 0..self.get_widget_height() {
             printer.print(
                 (0, lines),
-                &format!("{:0len$X}", lines * U8S_PER_LINE, len = digits_len),
+                &format!("{:0len$X}", lines *  self.config.bytes_per_line, len = digits_len),
             );
         }
     }
 
     fn draw_addr_hex_sep(&self, printer: &Printer) {
-        printer.print_vline((0, 0), self.get_widget_height(), ADDR_HEX_SEPERATOR);
+        printer.print_vline((0, 0), self.get_widget_height(),  &self.config.addr_hex_separator);
     }
 
     /// draws the hex fields between the addr and ascii representation
     fn draw_hex(&self, printer: &Printer) {
-        for (i, c) in self.data.chunks(U8S_PER_LINE).enumerate() {
+        let spacer = &format!("{:width$}", "", width=self.config.chars_per_spacing);
+        for (i, c) in self.data.chunks(self.config.bytes_per_line).enumerate() {
             let hex = c
-                .chunks(2)
+                .chunks(self.config.bytes_per_group)
                 .map(|c| {
-                    if c.len() == 2 {
-                        format!("{:02X}{:02X}", c[0], c[1])
-                    } else {
-                        format!("{:02X}", c[0])
+                    let mut s= String::new();
+                    for &b in c {
+                        write!(&mut s, "{:02X}", b).expect("Unable to write hex values");
                     }
+                    format!("{}", s)
```

Comment:
this can be simplified to `s.to_string()`

Reply:
Let me know if [this](https://github.com/palako/cursive_hexview/commit/b745a36a1b20b9b35c2a643615c00f6884b0ce1c) is good